iree/compiler/Dialect/VM/Conversion/VMToEmitC/ConvertVMToEmitC.cpp

```diff
@@ -1511,10 +1656,14 @@ class CallOpConversion : public OpConversionPattern<IREE::VM::CallOp> {
 
         if (failed(clearStruct(rewriter, refPtrOp.getResult(),
                                /*isPointer=*/true))) {
-          return failure();
+          return op.emitError() << "clearStruct call failed";
         }
 
-        bool move = ptr->second.isLastValueUse(operand, op.getOperation());
+        bool move = op.getOperation()
```

Comment:
yeah, this is not great - it's significantly slower than the analysis lookup and less readable and maintainable as now there's untyped deep access into string-based attribute storage: trying to change how the attributes are structured is going to be exceedingly difficult, and you aren't even going to be able to find it all without 100% test coverage as the compiler won't help you. These will also all segfault if the attributes are not precisely specified, but there's no way to put a verifier on these effectively and since this information is now carried in the IR structure no guarantee that other things won't happen to it before this runs.